includes/class-sensei-posttypes.php

```diff
@@ -233,6 +234,35 @@ public function setup_course_post_type() {
 
 	}
 
+
+	/**
+	 * Redirect visits to the page id specified by course page to the archive page of the course
+	 *
+	 * @since 4.0.2
+	 * @uses  Sensei()
+	 * @access private
+	 * @return void
+	 */
+	public function redirect_course_archive_page() {
+		$settings = Sensei()->settings->settings;
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+		// When default permalinks are enabled, redirect course page to post type archive url.
+		$course_url = get_post_type_archive_link( 'course' );
+		if (
+			! empty( $_GET['page_id'] ) &&
+			'' === get_option( 'permalink_structure' ) &&
+			absint( $_GET['page_id'] ) === absint( $settings['course_page'] ) &&
+			$course_url ) {
+			foreach ( $_GET as $param => $value ) {
+				if ( 'page_id' !== $param ) {
+					$course_url = add_query_arg( $param, $value, $course_url );
```

Comment:
hmm..I didn't added specialized escaping here because [`wp_safe_redirect()` already does sanitization and validation of the URL](https://developer.wordpress.org/reference/functions/wp_safe_redirect/), so it seemed enough to me, given that it [already does some sanitization and filtering of the value](https://developer.wordpress.org/reference/functions/wp_sanitize_redirect/).

I can escape the final `$course_url` using `esc_url()` without any issue, however, I'm not sure what to use to escape `$param` and `$value` in this case. Do you have any suggestions? Looking through the [list of escaping functions](https://developer.wordpress.org/themes/theme-security/data-sanitization-escaping/), but I didn't find any "perfect" options for this.

Reply:
That's correct! I think that sanitization from `wp_safe_redirect` is probably enough! =)